rfc/0018-certificate-handling.md

```diff
@@ -82,174 +86,199 @@ webhooks:
 
 - Generate new certificate, sign it with our CA
-- Create Policy Server Deployment, ensure the certificate is mounted inside of the Pods
+- Create Policy Server Deployment, ensure the certificate is mounted inside of
+  the Pods
 
-This is already done by our current code (Kubewarden 1.6.0).
+It's important to note that the CA used to sign the Policy Server certificate
+is the same CA used to sign the `kubewarden-controller` certificate. It's not
+the internal CA generated to signed Policy Server certificates as it was done
+before.
 
-### New `AdmissionPolicy`/`ClusterAdmissionPolicy` is defined
+### New
+`AdmissionPolicy`/`ClusterAdmissionPolicy`/`ClusterAdmissionPolicyGroup`/`AdmissionPolicyGroup`
+is defined
 
-A new policy is defined inside of the cluster, ensure the following action are done:
+A new policy is defined inside of the cluster, ensure the following action are
+done:
 
-- Register the webhook against the Kubernetes API, use our CA inside of `clientConfig.caBundle`
-
-> **Note:** it doesn't matter which Policy Server is going to host the policy. All the Policy Server
-> instances use a certificate that is signed by our CA. Hence we always put our CA
-> inside of the `clientConfig.caBundle`.
+- Register the webhook against the Kubernetes API, use our CA inside of
+  `clientConfig.caBundle`
 
-This is already done by our current code (Kubewarden 1.6.0).
+> **Note:** it doesn't matter which Policy Server is going to host the policy.
+> All the Policy Server instances use a certificate that is signed by our CA.
+> Hence we always put our CA inside of the `clientConfig.caBundle`.
 
-### Policy Server Certificate Renewed
+### Policy Server and `kubewarden-controller` Certificates Renewed
 
-From time to time we need to renew the certificate issued to a Policy Server instance. The main reason
-to perform this operation is to ensure the Policy Server is not using an expired certificate.
+From time to time we need to renew the certificate issued to the
+`kubewarden-controller` and a Policy Server instance. The main reason to
+perform this operation is to ensure the certicates in use are not expired.
 
-To renew a Policy Server certificate the following actions have to be done:
+To renew a certificate the following actions have to be done:
 
 - Generate new certificate, sign it with our CA
-- Force a rollout of the Policy Server deployment. This will ensure the HTTPS server uses the new certificate
-
-> **Note:** there's no need to touch the webhook configuration objects related to the policies
-> hosted by the Policy Server. The new certificate is still signed by the same CA, hence nothing has
+- Update the secret storing the certificate
+- Kubernetes automatically updates the certificate secret mounted inside of the
+  Pods that are running the Policy Server or the `kubewarden-controller`. Both
+  types of running process will notice the change of the certificate file they
+  read at boot time and will reload their TLS configuration.
```

Comment:
not at boot time as we are auto-realoading in both components

Reply:
Yes, that's what the sentence is trying to say in a different way. Let me rephrase it to make it more clear.